microsoft-edge/extensions-chromium/developer-guide/sidebar.md

```diff
@@ -0,0 +1,625 @@
+
+`manifest.json`:
+
+```json
+{
+  "name": "My sidebar extension",
+  ...
+  "side_panel": {
+    "default_path": "sidebar.html"
+  }
+  ...
+}
+```
+
+`sidebar.html`:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>My Sidebar</title>
+  </head>
+  <body>
+    <h1>Sidebar extension for all sites</h1>
+    <p>This sidebar is enabled on all sites</p>
+  </body>
+</html>
+```
+
+
+<!-- ------------------------------ -->
+#### Enable a sidebar for a specific site only
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#by-site -->
+
+An extension can use [sidepanel.setOptions()](https://developer.chrome.com/docs/extensions/reference/sidePanel/#method-setOptions) to enable a sidebar on a specific tab.
+This can be a particular website a user is using and would want the same extension to open in the sidebar, once this website is opened.
+
+This example uses [chrome.tabs.onUpdated()](https://developer.chrome.com/docs/extensions/reference/tabs/#event-onUpdated) to listen for any updates made to the tab.  This example checks whether the URL is `www.bing.com` and conditionally enables the sidebar.  Otherwise, this example disables the sidebar.
+
+`service-worker.js`:
+
+```js
+const BING_ORIGIN = 'https://www.bing.com';
+
+chrome.tabs.onUpdated.addListener(async (tabId, info, tab) => {
+  if (!tab.url) return;
+  const url = new URL(tab.url);
+  // Enables the sidebar when at bing.com
+  if (url.origin === BING_ORIGIN) {
+    await chrome.sidePanel.setOptions({
+      tabId,
+      path: 'sidepanel.html',
+      enabled: true
+    });
+  } else {
+    // Disables the sidebar when at other sites
+    await chrome.sidePanel.setOptions({
+      tabId,
+      enabled: false
+    });
+  }
+});
+```
+
+When a user switches to a tab or site for which the sidebar is not enabled, the sidebar is hidden.
+
+For a complete example, see the [Site-specific side panel example](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/cookbook.sidepanel-site-specific) sample.
+
+
+<!-- ------------------------------ -->
+#### Enable the extension's shortcut icon to open the sidebar
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#open-action-icon -->
+
+To allow users to open the sidebar by clicking the action toolbar icon, use [sidePanel.setPanelBehavior()](https://developer.chrome.com/docs/extensions/reference/sidePanel/#method-setPanelBehavior).  First, declare the `"action"` key in the manifest:
+
+`manifest.json`:
+
+```json
+{
+  "name": "My sidebar extension",
+  ...
+   "action": {
+    "default_title": "Click to open sidebar"
+  },
+  ...
+}
+```
+
+Then, add the following code to the code listing that's in [Enable a sidebar for a specific site only](#enable-a-sidebar-for-a-specific-site-only) above:
+
+`service-worker.js`:
+
+```js
+const BING_ORIGIN = 'https://www.bing.com';
```

Comment:
The `BING_ORIGIN` const is not needed here.

Reply:
comparing https://developer.chrome.com/docs/extensions/reference/sidePanel/#open-action-icon ...